mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -28,10 +30,26 @@ object MvRxViewModelProvider {
         stateClass: Class<out S>,
         viewModelContext: ViewModelContext,
         key: String = viewModelClass.name,
-        initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory()
+        initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory(),
```

Comment:
does `initialStateFactory` need to be a parameter? seems like it is only used in the `viewModelFactory` default value

Reply:
this was actually added in https://github.com/airbnb/MvRx/pull/219 as we override this in the Airbnb app for mocking

Reply:
Right, we need to make sure we can override the necessary behavior or mocking, but we also want a clean interface here.

As far as I can tell the only times we override `viewModelFactory` are when we want to throw if the view model doesn't already exist - ie factory that throws when it is used instead of actually doing anything. perhaps we don't need to allow `viewModelFactory` to be set, and instead this can have a boolean parameter for whether it should throw if the viewmodel doesn't already exist

we should try to have a clean interface here and not need both `initialStateFactory` and `viewModelFactory`, but we do need to design it in a way that works for our mocking (ie it needs to be overridable